src/query/api/v1/middleware/metrics.go

```diff
@@ -71,11 +93,12 @@ type routeMetrics struct {
 	sync.RWMutex
 	instrumentOpts instrument.Options
 	metrics        map[routeMetricKey]routeMetric
-	timers         map[string]tally.Timer
+	timers         map[routeMetricKey]tally.Timer
```

Comment:
Should we just add the timer to the “routeMetric” struct now that it’s keyed by the same thing as all the other metrics? We only had this separately since it was keyed by path and not status as well.

Reply:
The idea here is this is only optionally keyed by status; i.e. if we have `cfg.AddStatusToLatencies` set to `false`, which will likely be the majority of non-debug cases, we won't be keying timers by latency, or adding the code tag to them: i.e. two requests, one of which has `status = 200` and the other `status = 300`, will increment the same timer if we're not adding status here. Conversely, we always want to track the `status` field for regular metrics, so the two have slightly different paths